### PR TITLE
Cluster-wide configuration for query vectorization

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

We currently should put some configurations in the query context per query to enable query vectorization. This PR is to add a cluster-wide configuration for query vectorization.

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/incubator-druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/incubator-druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

---
This change is [Reviewable](https://reviewable.io/reviews/apache/incubator-druid/8657)

### CODE REVIEW

**Comment by jihoonson** [comment]
@gianm thanks, fixed.

**Comment by gianm** [comment]
@jihoonson there's a few `ImportOrder` style twoggles.

**Comment by jihoonson** [comment]
@gianm oh sorry, it was because of the rename. Fixed now.

File: processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java [hunk 1]
```diff
@@ -98,6 +100,7 @@
   public GroupByStrategyV2(
       DruidProcessingConfig processingConfig,
       Supplier<GroupByQueryConfig> configSupplier,
+      Supplier<QueryConfig> vectorizationConfigSupplier,
```
Comment (gianm) [inline processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java]: Should be `queryConfigSupplier`?

File: processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java [hunk 2]
```diff
@@ -89,6 +90,7 @@
 
   private final DruidProcessingConfig processingConfig;
   private final Supplier<GroupByQueryConfig> configSupplier;
+  private final Supplier<QueryConfig> vectorizationConfigSupplier;
```
Comment (gianm) [inline processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java]: `queryConfigSupplier`?

File: processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryEngine.java [hunk 3]
```diff
@@ -57,6 +59,7 @@
  */
 public class TimeseriesQueryEngine
 {
+  private final Supplier<QueryConfig> vectorizationConfigSupplier;
```
Comment (gianm) [inline processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryEngine.java]: Same, should be `queryConfigSupplier`.

File: processing/src/test/java/org/apache/druid/query/groupby/GroupByLimitPushDownInsufficientBufferTest.java [hunk 4]
```diff
@@ -352,8 +353,10 @@ public String getFormatString()
       }
     };
 
-
     final Supplier<GroupByQueryConfig> configSupplier = Suppliers.ofInstance(config);
+    final Supplier<QueryConfig> vectorizationConfigSupplier = Suppliers.ofInstance(
```
Comment (gianm) [inline processing/src/test/java/org/apache/druid/query/groupby/GroupByLimitPushDownInsufficientBufferTest.java]: `queryConfigSupplier`?

I'm not gonna point out the rest :P

File: processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java (hunk 1 above)
Comment (jihoonson) [reply processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java]: Oops, fixed all variable names.

**Pull request merged**